lib/spack/spack/spec.py

```diff
@@ -4567,6 +4567,9 @@ def install_status(self):
         """Helper for tree to print DB install status."""
         if not self.concrete:
             return None
+
+        if self.external:
+            return "external"
```

Comment:
I know Python doesn't have enums, but sure there are better things than a function returning the union of `str`, `bool` and `None` and giving meaning to those. Can you please use something decent? E.g.

```python
class InstallStatus:
    EXTERNAL = 0
    INSTALLED_UPSTREAM = 1
    INSTALLED_LOCALLY = 2
    NOT_INSTALLED = 3
```

Also, `get_record` calls `query_by_spec_hash(key)` which is also called later for upstreams, sounds unnecessary to me.

Reply:
https://docs.python.org/3/library/enum.html

Reply:
There's also https://pypi.org/project/enum34/ that is supported in earlier versions (including 2.7).

Reply:
So that's something we can use in 0.19 🙃